scripts/schema-publish.sh

```diff
@@ -6,15 +6,38 @@
 
 schemaDir="src/schemas/validation"
 branch=$(git branch --show-current)
-version=${branch:1:3}
-echo === Building schemas into ./deploy/oas/$version
+
+if [[ $branch =~ ^v([0-9]+\.[0-9]+\.[0-9]+) ]]; then
+  deploydir="./deploy/oas/${BASH_REMATCH[1]}"
+else
+  deploydir="./deploy-preview"
+fi
+
+# create the date-stamped schemas
+publish_schema() {
+  local schema="$1"
+  local date="$2"
+
+  local base=$(basename $schema '.yaml')
+  local target=$deploydir/$base/$date
+
+  mkdir -p $deploydir/$base
+  # replace the WORK-IN-PROGRESS placeholder
+  sed -e "s/${base}\\/WORK-IN-PROGRESS/${base}\\/${date}/g" $schemaDir/$schema > $target.yaml
```

Comment:
This is insufficient as the four schema files reference each other, and these references also need to be updated with the date for the corresponding referenced schema file, for example in 

https://github.com/OAI/OpenAPI-Specification/blob/9fd4e2ebb49625f407e176ab507e2ca693789a5c/src/schemas/validation/dialect.yaml#L21

Reply:
Thanks. I pushed a change to address this by accumulating sed sub-commands in an array and serializing the full sed script as an argument to `publish_schema`.